ribs/emitters/rankers.py

```diff
@@ -0,0 +1,368 @@
+             measure_batch, metadata, add_statuses, add_values):
+        # New solutions sort ahead of improved ones, which sort ahead of ones
+        # that were not added. Note that lexsort sorts the values in ascending
+        # order, so we use numpy fancy indexing to reverse the sorted array.
+        return np.lexsort((add_values, add_statuses))[::-1]
+
+    # Generates the docstring for rank
+    rank.__doc__ = f"""
+Generates a list of indices that represents an ordering of solutions.
+
+Args:
+{_args.emitter}
+{_args.archive}
+{_args.solution_batch}
+{_args.objective_batch}
+{_args.measure_batch}
+{_args.metadata}
+{_args.add_statuses}
+{_args.add_values}
+
+Returns:
+{_returns.indices}
+    """
+
+
+class RandomDirectionRanker(RankerBase):
+    """Ranks the solutions based on projection onto a direction in
+    measure space.
+
+    This ranker originates in `Fontaine 2020
+    <https://arxiv.org/abs/1912.02400>`_ as RandomDirectionEmitter.
+    The solutions are ranked solely based on their projection onto a random
+    direction in measure space.
+
+    To rank the solutions first by whether they were added, and then by
+    the projection, refer to
+    :class:`ribs.emitters.rankers.TwoStageRandomDirectionRanker`.
+    """
+
+    def __init__(self, seed=None):
+        self._target_measure_dir = None
+        self._rng = np.random.default_rng(seed)
+
+    @property
+    def target_measure_dir(self):
+        """numpy.ndarray: ``(measure_dim,)`` array with the target measure
+        direction vector."""
+        return self._target_measure_dir
+
+    @target_measure_dir.setter
+    def target_measure_dir(self, value):
+        self._target_measure_dir = value
+
+    def rank(self, emitter, archive, solution_batch, objective_batch,
+             measure_batch, metadata, add_statuses, add_values):
+        projections = np.dot(measure_batch, self._target_measure_dir)
+        # Sort only by projection; use fancy indexing to reverse the order
+        return np.lexsort((projections,))[::-1]
+
+    # Generates the docstring for rank
+    rank.__doc__ = f"""
+Ranks the soutions based on projection onto a direction in behavior space.
+
+Args:
+{_args.emitter}
+{_args.archive}
+{_args.solution_batch}
+{_args.objective_batch}
+{_args.measure_batch}
+{_args.metadata}
+{_args.add_statuses}
+{_args.add_values}
+
+Returns:
+{_returns.indices}
+    """
+
+    def reset(self, emitter, archive):
+        ranges = archive.upper_bounds - archive.lower_bounds
+        behavior_dim = len(ranges)
+        unscaled_dir = self._rng.standard_normal(behavior_dim)
+        self._target_measure_dir = unscaled_dir * ranges
+
+    # Generates the docstring for reset
+    reset.__doc__ = f"""
+Generates a new random direction in the behavior space.
+
+The direction is sampled from a standard Gaussian -- since the standard
+Gaussian is isotropic, there is equal probability for any direction. The
+direction is then scaled to the behavior space bounds.
```

Comment:
Worth mentioning that we scale by ranges so that we select a random _archive direction_, not just a random direction in space.

Reply:
should we change `target_behavior_dir` to `target_archive_dir`?